torch/linalg/__init__.py

```diff
@@ -1629,6 +1629,25 @@
   Please use :func:`torch.linalg.svdvals`, which computes only the singular values,
   instead of `compute_uv=False`.
 
+.. note:: Selection of `driver` (this keyword argument only works on CUDA inputs with cuSOLVER backend):
+
+    - When `driver` is not given or set to `None` (*default*), a driver will be selected automatically
+      based on the shape of the inputs.
+      The default choice should be efficient and precise for most inputs. However, if input matrices
+      have large condition numbers or could be ill-conditioned, it's recommended to choose the `gesvd` driver
+      or use higher precision data type (e.g. use double instead of float).
+    - The `gesvd` driver directly calls the cuSOLVER `cusolverDn<t>gesvd` method. This QR-based algorithm is
+      less performant on GPU, but it can converge ill-conditioned matrices and get more precise results for those inputs
+      than the other drivers.
+    - The `gesvdj` driver calls the cuSOLVER `cusolverDn<t>gesvdjBatched` method if input batched matrices
+      are square matrices with sizes less than or equal to 32. Otherwise, the `gesvdj` driver calls the cuSOLVER
+      `cusolverDn<t>gesvdj` method. This Jacobi-based algorithm has better performance than `gesvd` on GPU,
+      but it may get less accurate results for ill-conditioned matrices or matrices with very large sizes
+      (be cautious if matrix size > 2048).
+    - The `gesvda` driver directly calls the cuSOLVER `cusolverDn<t>gesvdaStridedBatch` method. This is an
+      approximation method that is preferred when the number of rows is much different from the number of columns
+      (`m >> n` or `n >> m`).
+
```

Comment:
I thought that, in the interest of conciseness and uniformity, it could be good to do as we do in `linalg.lstsq` and provide the user with a guide on how to choose the backend. What about rather than writing this note, we describe it as any other argument before the "differences with `np.linalg.svd`" section?
```
The `driver` kwarg may be used in CUDA with a cuSOLVER backend to choose the algorithm used to compute the SVD.
The choice of a driver is a trade-off between accuracy and speed.

- If :attr:`A` is well-conditioned (its `condition number`_ is not too large), or you do not mind some precision loss.

  - For a general matrix: `'gesvdj'` (Jacobi method)
  - If :attr:`A` is tall or wide (`m >> n` or `m << n`): `'gesvda'` (Approximate method)

- If :attr:`A` is not well-conditioned or precision is relevant: `'gesvd'` (QR based)

By default (:attr:`driver`\ `= None`), we call `'gesvdj'` and, if it fails, we fallback to `'gesvd'`.
```

Reply:
Done

Reply:
My local doc build is broken, gonna use the CI to check how the updated doc looks like.

Reply:
without the underscore after `condition number`

![image](https://user-images.githubusercontent.com/24860335/168976136-16d5e09f-d299-41ad-b082-a7c2e85ad866.png)

Reply:
Sorry, this is missing a link at the end of the docs of the form
```
.. _condition number:
    https://pytorch.org/docs/master/linalg.html#torch.linalg.cond
```
Check the docs for `linalg.lstsq`, as this is inspired by those.
Also, could you put ths, rather than in a note, in plain text before the section "differences with np.linalg.svd"? This is a description of an input to the function, so it should go in the main body of the docs.